### ISSUE TITLE
Locking Tabs Possible?

### ISSUE THREAD

**Issue by LookingForWayOut** [piroor/treestyletab#3014]
Hello. Question.

There are so many times I arrange my tabs in a nice tree order I prefer, but then sometimes something with Firefox or another addon will move my tabs all over the place. I then have to rearrange. It happens randomly, usually when restarting Firefox but not every time. However, if I use a program that sorts all tabs in an order, like alphabetical/etc, then they all get moved, yes.

So, using that example, is there a way to lock the tabs they way they are from being moved without user interaction? Is there a way for a tree to stay as a tree, unless the user moves the tab or closes it?

Thank you.

**Comment by irvinm** [comment]
I'm not positive on this, but in general you have to remember that TST is just a virtual representation of the native tabs in Firefox.  Those native tabs can be manipulated by Firefox and other addons and TST can not prevent that.  (I guess there could be an addon to monitor tabs and when they change to try to determine if they should be put back ... but I doubt there is enough information as to the nature of the change whether it was initiated by a user or by Firefox or an addon)

So, long story short, I don't think this is possible ... and certainly not by TST currently.  @piroor correct me if I am wrong.

**Comment by LookingForWayOut** [comment]
That makes sense. Was a thought I had.
Thank you.

**Comment by irvinm** [comment]
@piroor #WontFix, #WorksAsDesigned, #Invalid

**Comment by piroor** [comment]
@irvinm Thank you to explain TST's background!

@LookingForWayOut As described above this beyonds TST's purpose, thus I won't implement the option to TST.

On the other hand, TST actually cancels tab move on some case: a tab move produced by someone (for example a drag and drop in the horizontal tab bar) breaking the tree structure is automatically reverted. I think you can create an extension providing the behavior, and if you create it as a helper extension of TST, TST may work with it together well via its API: https://github.com/piroor/treestyletab/wiki/API-for-other-addons . Sorry I won't implement such an extension by myself, but I'm welcome to make TST compatible with such an extension or extending TST's API for such an extension if needed.
